### ISSUE TITLE
Support multiple scripts in a language

### ISSUE THREAD

**Issue by mterencelao** [hugolabe/Wike#23]
Some languages in Wikipedia can be displayed in multiple scripts. In Serbian, for instance, Wikipedia offers an option to display Serbian articles in Latin or Cyrillic. Likewise for Chinese, which has Traditional (for Hong Kong, Macau, and Taiwan) and Simplified (chiefly Mainland China) scripts. It would be great if we could read articles exclusively in our chosen script for such languages.

P.S. I would have asked if this app could be more than just a Wikipedia reader, since it's built on MediaWiki, but I could wholly bet that will never happen. ✌

**Comment by hugolabe** [comment]
I was not familiar with this subject of scripts in some languages.

From what I've seen, on Wikipedia's mobile web they appear in the top of the list of available languages ​for the article. The problem is that Wike hides this list and uses the API to get it.

Another problem is that the URLs of these variants are not the standard Wikipedia ones (eg sr.wikipedia.org/sr-el /... instead of sr.wikipedia.org/wiki /...) and Wike checks them to find out whether to open a link in the app or in the browser.

It's a somewhat complicated subject. I will try to do some more research.

**Comment by mterencelao** [comment]
Looks like the situation's really more complicated than I thought. I did my own investigation as well, trying a couple other languages known to have multiple scripts (Kazakh, Kurdish, ...), and it turns out Wikipedia has another pattern for variants other than the one you pointed: <locale>.wikipedia.org/w/index.php?title=<article>&variant=<script>

Thanks for pointing out the _proper name_ for this situation. I'll revise the title of this issue.

However, looking at Wikipedia's official mobile app (for Android, at least), they explicitly list down the variants in a separate file. See https://github.com/wikimedia/apps-android-wikipedia/blob/main/app/src/main/res/values/languages_list.xml

**Comment by hugolabe** [comment]
Yes. There are two [alternative URLs](https://en.wikipedia.org/wiki/Help:URL#URLs_of_Wikipedia_pages) to access Wikipedia pages. Wike uses the first one, which is the one returned by the search results and used in the wikilinks.

What I have not seen is that through the API you can obtain the links to the variants of a language. I think it would have to be done by manipulating the URL manually.

I'll be doing some tests when I have time.

Thanks for the link to the Android languages list. It helps me.

**Comment by null-von-sushi** [comment]
Any progress on this?
For Chinese at least, loading the traditional Chinese version used in Taiwan would simply mean replacing the `wiki` part of the URL with `zh-tw`. 
 
As for links on the website, some links do indeed link to variant pages, but most links seem to target /wiki/name_of_article, which is automatically redirected to the last-selected variant. The app uses HTML to render wikipedia, right? If so, just adding a button in the settings that temporarily loads a normal wikipedia page in-app to let the user select the preference, and wikipedia should remember the preference for any future pages loaded in that language. Unless the app doesn't store cookies?

**Comment by hugolabe** [comment]
Sorry but no. The time I can dedicate to the application is limited and I have prioritized other issues.

> For Chinese at least, loading the traditional Chinese version used in Taiwan would simply mean replacing the `wiki` part of the URL with `zh-tw`.

It's the same in other languages ​​with variants. This is a problem, because Wike parses the URL of the page and if it doesn't contain '/wiki/' in the path, it doesn't consider it a valid article and opens it in the browser.
